aachen.py

```diff
@@ -1,79 +1,118 @@
-    # 1. menues
-    note = data.find(id='note')
-    if note:
+    if is_closed(data):
         canteen.setDayClosed(day)
         return
-    add_meals_from_table(canteen, data.find(attrs={'class': 'menues'}), day)
 
-    # 2. extras:
-    extras_table = data.find(attrs={'class': 'extras'})
+    meals_table = data.find(attrs={'class': 'menues'})
+    add_meals_from_table(canteen, meals_table, day)
 
-    if not extras_table:
-        return
+    extras_table = data.find(attrs={'class': 'extras'})
     add_meals_from_table(canteen, extras_table, day)
 
 
-def parse_url(url, today=False):
-    canteen = OpenMensaCanteen()
-    # todo only for: Tellergericht, vegetarisch, Klassiker, Empfehlung des Tages:
-    canteen.setAdditionalCharges('student', {'other': 1.5})
+def is_closed(data):
+    note = data.find(id='note')
+    if note:
+        return True
+    else:
+        return False
 
-    document = parse(urlopen(url).read(), 'lxml')
 
-    global legend
-    regex = '\((?P<name>[\dA-Z]+)\)\s*(?P<value>[\w\s]+)'
-    legend = buildLegend(legend, document.find(id='additives').text, regex=regex)
+def add_meals_from_table(canteen, table, day):
+    for item in table.find_all('tr'):
+        category, name, notes, price_tag = parse_meal(item, canteen.legend)
+        canteen.addMeal(day, category, name, notes, prices=price_tag)
+
+
+def parse_meal(table_row, legend):
+    category = table_row.find('span', attrs={'class': 'menue-category'}).text.strip()
+
+    description_container = table_row.find('span', attrs={'class': 'menue-desc'})
+    clean_description_container = get_cleaned_description_container(description_container)
+    name, notes = parse_description(clean_description_container, legend)
+
+    price_tag = table_row.find('span', attrs={'class': 'menue-price'})
+    if price_tag:
+        price_tag = price_tag.text.strip()
+
+    return category, name, notes, price_tag
+
+
+def get_cleaned_description_container(description_container):
+    # "Hauptbeilage" and "Nebenbeilage" are flat, while the others are wrapped in <span class="expand-nutr">
+    effective_description_container = description_container.find('span', attrs={
+        'class': 'expand-nutr'}) or description_container
+
+    def is_valid_description_element(element):
+        if not isinstance(element, Tag):
+            return True
+        # Keep <span class="seperator">oder</span>
+        if element.name == 'span' and 'seperator' in element['class']:
+            return True
+        return False
+
+    description_container = list(filter(is_valid_description_element, effective_description_container.children))
+
+    return description_container
+
+
+def parse_description(description_container, legend):
+    name_parts = []
+    notes = set()
+    for element in description_container:
+        if type(element) is Tag and element.name == 'sup':
+            notes.update(element.text.strip().split(','))
+        else:
+            name_parts.append(element.string.strip())
```

Comment:
calling `parser.parse("aachen", "vita", None))` this crashes here for me, end of backtrace:
~~~
  File "openmensa-parsers/aachen.py", line 104, in parse_description
    name_parts.append(element.strip())
TypeError: 'NoneType' object is not callable